### PR TITLE
Optimize Pricefeed EndBlocker

### PR DESCRIPTION
**Pull request by nddeluca**
Added tests that cover 100% of legacy behavior (see TestKeeper_SetCurrentPrices_MatchesAllMarketsBehavior), and added an optimized version of set current prices for all markets.

### CODE REVIEW

File: x/pricefeed/keeper/keeper.go [hunk 1]
```diff
@@ -131,6 +131,74 @@ func (k Keeper) SetCurrentPrices(ctx sdk.Context, marketID string) error {
...
+		}
+
+		notExpiredPrices, _ := marketPricesByID[marketID]
+
+		if len(notExpiredPrices) == 0 {
+			// NOTE: The current price stored will continue storing the most recent (expired)
+			// price if this is not set.
+			// This zero's out the current price stored value for that market and ensures
+			// that CDP methods that GetCurrentPrice will return error.
+			k.setCurrentPrice(ctx, marketID, types.CurrentPrice{})
```
Comment (nddeluca) [inline x/pricefeed/keeper/keeper.go:178]: Verified that removing this line fails tests.

File: x/pricefeed/keeper/keeper.go [hunk 2]
```diff
@@ -131,6 +131,74 @@ func (k Keeper) SetCurrentPrices(ctx sdk.Context, marketID string) error {
...
+			continue
+		}
+
+		medianPrice := k.CalculateMedianPrice(notExpiredPrices)
```
Comment (nddeluca) [inline x/pricefeed/keeper/keeper.go:182]: Tests fail if mean is used instead of median

File: x/pricefeed/keeper/keeper.go [hunk 3]
```diff
@@ -131,6 +131,74 @@ func (k Keeper) SetCurrentPrices(ctx sdk.Context, marketID string) error {
...
+
+		// check case that market price was not set in genesis
+		//if validPrevPrice && !medianPrice.Equal(prevPrice.Price) {
+		if validPrevPrice && !medianPrice.Equal(prevPrice.Price) {
```
Comment (nddeluca) [inline x/pricefeed/keeper/keeper.go:186]: Tested event emission only occurs when this logic statement is true

**Review by galxy25 — APPROVED** [review]

**Comment by nddeluca** [comment]
Merging to master, but not back-porting to v0.26.x until after release

**Pull request merged**